FinanceTracker/FinanceTracker/Utils/UserInputReader.cs

```diff
@@ -1,15 +1,42 @@
 ﻿using FinanceTracker.Enums;
+using FinanceTracker.Models;
 
 namespace FinanceTracker.Utils
 {
     public class UserInputReader
     {
+        private const string FILEPATH = "categories.json";
+        public int GetCategory(OperationType type)
+        {
+            List<Category> categories = JsonFileManager.ReadCategoriesFromJson(FILEPATH);
+            categories = categories.Where(category => category.Type == type).ToList();
+
+            bool validEntry = false;
+            int categoryId = 0;
+            string userInput;
+            const string displayMessage = "Enter a category from listed below:";
+
+            while (!validEntry)
+            {
+                userInput = GetCategoryChoice(displayMessage, categories).ToLower();
+
+                var matchingCategory = categories.FirstOrDefault(category => category.Name?.ToLower() == userInput);
+
+                if (matchingCategory != null)
+                {
+                    categoryId = matchingCategory.Id;
```

Comment:
You can `return matchingCategory.Id` right here and you will not need to use `validEntry` at all